### PR TITLE
Conditionally enable AVX2 256-bit instructions

### PR DESCRIPTION
**Pull request by sfiligoi**
Most recent CPUs support AVX2 256-bit vector instructions, but SwAligner was limited to 128-bit SSE2.
With the change, the SwAligner method can obtain the same result with half the instructions.

Note: The AVX2 code path is still disabled by default, but can be enabled by setting the SSE_AVX2 env variable at compile time.

### CODE REVIEW

**Comment by sfiligoi** [comment]
@ch4rr0 As usual, looks like the test-random job hit the wallclock limit; I don't see any errors.
The code should be ready for review.

**Comment by sfiligoi** [comment]
@ch4rr0 I took the conservative approach of not enabling the AVX2 code by default.
After this is accepted and merged, we should discuss if and how to make it part of the default distribution binary.

**Comment by ch4rr0** [comment]
> @ch4rr0 As usual, looks like the test-random job hit the wallclock limit; I don't see any errors.
> 
> The code should be ready for review.

Hello, I'm going to separate the simple and random tests into separate workflows. Hopefully it solves this issue.

**Comment by ch4rr0** [comment]
> @ch4rr0 I took the conservative approach of not enabling the AVX2 code by default.
> 
> After this is accepted and merged, we should discuss if and how to make it part of the default distribution binary.

Sounds good.

**Comment by sfiligoi** [comment]
> > @ch4rr0 As usual, looks like the test-random job hit the wallclock limit; I don't see any errors.
> > The code should be ready for review.
> 
> Hello, I'm going to separate the simple and random tests into separate workflows. Hopefully it solves this issue.

Note: I already separate the two.
But I do re-compile, too.
Let me try to just reuse the binaries.

**Comment by sfiligoi** [comment]
> @ch4rr0 As usual, looks like the test-random job hit the wallclock limit; I don't see any errors.
> The code should be ready for review.
FYI: The error is real, the error message is just hidden deep in the text.
Working on a fix.

**Comment by sfiligoi** [comment]
@ch4rr0 Success. All tests are passing now.

**Comment by sfiligoi** [comment]
@ch4rr0 Anything else that needs changing before it can be merged?

**Comment by sfiligoi** [comment]
Just a reminder that I do not have write privileges, and someone else needs to either merge or reject the PR.

**Comment by ch4rr0** [comment]
Hello @sfiligoi, I was on vacation hence the lapse in communication. Ben and I will be reviewing the changes. We appreciate the time spent putting this PR together.

File: sse_wrap.h [hunk 1]
```diff
@@ -27,41 +27,220 @@
...
+
+typedef __m256i SSERegI;
+#define sse_adds_epi16(x, y) _mm256_adds_epi16(x, y)
+#define sse_adds_epu8(x, y) _mm256_adds_epu8(x, y)
+#define sse_cmpeq_epi16(x, y) _mm256_cmpeq_epi16(x, y)
+#define sse_cmpeq_epi8(x, y) _mm256_cmpeq_epi8(x, y)
+#define sse_cmpgt_epi16(x, y) _mm256_cmpgt_epi16(x, y)
+#define sse_cmpgt_epi8(x, y) _mm256_cmpgt_epi8(x, y)
+#define sse_cmplt_epi16(x, y) _mm256_cmpgt_epi16(y,x)
+#define sse_extract_epi16(x, y) _mm256_extract_epi16(x, y)
```
[the remark this answers is not in the document]
Comment (sfiligoi) [reply sse_wrap.h:44]: No, there is no cmplt variant in AVX2.
Actually, there is no hardware instruction for cmplt in SSE, either, but the header provides a handy wrapper that switches the arguments there (but was dropped in AVX headers).

**Pull request merged**